### DISCUSSION TITLE
Python Function fails if the json response fields are not as expected

### DISCUSSION THREAD

**Discussion by eternalparquet** [appwrite/appwrite#5963] [Q&A]
Hello, I deployed a cloud function in python.
I deployed succesfully but when I execute it and the json data fields are not as
```
return res.json({
    'message': 'Hello from Appwrite!',
    'payload': payload,
    'secretKey': secretKey,
    'randomNumber': randomNumber,
    'trigger': trigger,
  })
```
it fail the execution with "Internal error" and there are not any other messages.
This happens also when I try to store a json string with json.dumps in the message field or payload value is not what I've sent in the request.
Is it normal or are there issue to be fixed?

**Comment by stnguyen90** [comment]
This typically happens if your code throws an exception. We're improving error handling, but, for now, I recommend:

1. wrapping your code in try/except to ensure your code does not throw any exception
2. ensuring you call `res.send()` or `res.json()` exactly once before exiting your function. 
3. don't send any custom status code

If you need any help, please create a support post on [Discord](https://appwrite.io/discord).
